### DISCUSSION TITLE
Problem with maps of two activities

### DISCUSSION THREAD

**Discussion by alich65** [mapsforge/vtm#894] [Q&A] [answered]
Hi,
I have two activities with map. When I back to MainActivity, the map not showing. Please watch below video. This is my code:

**MainActivity.java**

    `public class MainActivity extends AppCompatActivity {

    private MapView mapView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btnSecond = findViewById(R.id.btn_second_activity);
        btnSecond.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), SecondActivity.class);
                startActivity(intent);
            }
        });

        createMap();

        addBaseMap();
    }

    private void createMap(){
        mapView = findViewById(R.id.map_view_main);

        mapView.setClickable(true);

        GeoPoint center = new GeoPoint(47.7734, 10.0408);

        MapPosition centerMap = new MapPosition(center.getLatitude(), center.getLongitude(),50);
        mapView.map().animator().animateTo(centerMap);
    }

    private void addBaseMap(){
        BitmapTileSource osmTileSource = BitmapTileSource.builder()
                .url("https://tile.openstreetmap.org")
                .tilePath("/{Z}/{X}/{Y}.png")
                .build();

        BitmapTileLayer osmLayer = new BitmapTileLayer(mapView.map(), osmTileSource);
        osmLayer.setEnabled(true);
        mapView.map().layers().add(osmLayer);
    }

    @Override
    protected void onResume() {
        super.onResume();
        mapView.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mapView.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mapView.onDestroy();
    }
    }`

**SecondActivity.java**

         `public class SecondActivity extends AppCompatActivity {

    private MapView mapView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);

        createMap();

        addBaseMap();
    }

    private void createMap(){
        mapView = findViewById(R.id.map_view_second);
        mapView.setClickable(true);

        GeoPoint center = new GeoPoint(47.7734, 10.0408);

        MapPosition centerMap = new MapPosition(center.getLatitude(), center.getLongitude(),200);
        mapView.map().animator().animateTo(centerMap);
    }

    private void addBaseMap(){
        BitmapTileSource osmTileSource = BitmapTileSource.builder()
                .url("https://tile.openstreetmap.org")
                .tilePath("/{Z}/{X}/{Y}.png")
                .build();

        BitmapTileLayer osmLayer = new BitmapTileLayer(mapView.map(), osmTileSource);
        osmLayer.setEnabled(true);
        mapView.map().layers().add(osmLayer);
    }

    @Override
    protected void onResume() {
        super.onResume();
        mapView.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        mapView.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mapView.onDestroy();
    }

    }`

Can anyone help me?
Thanks

https://user-images.githubusercontent.com/40024574/147828029-bf77e2b8-d7c8-4e77-9b36-29e663b996f7.mp4

**Comment by devemux86** [comment, marked as answer]
Multiple VTM instances are not supported (#888).

However you can always return to a working map view.
See how [vtm-android-example](https://github.com/mapsforge/vtm/tree/master/vtm-android-example) samples work and [BitmapTileActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/BitmapTileActivity.java) example.

**Reply by alich65** [reply]
Is there any way for do this?
Can I use mapView's Mapsforge  in my VTM project?

**Reply by devemux86** [reply]
Mapsforge [MapView](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-android/src/main/java/org/mapsforge/map/android/view/MapView.java) is a regular View, so you can place it in any layout.
